frontend/src/components/chat/ChatInput.tsx

```diff
@@ -34,39 +47,59 @@ function ChatInput({ disabled, onSendMessage }: ChatInputProps) {
   };
 
   return (
-    <div className="w-full relative text-base flex">
-      <Textarea
-        value={message}
-        onChange={(e) => setMessage(e.target.value)}
-        onKeyDown={onKeyPress}
-        onCompositionStart={() => setIsComposing(true)}
-        onCompositionEnd={() => setIsComposing(false)}
-        placeholder={t(I18nKey.CHAT_INTERFACE$INPUT_PLACEHOLDER)}
-        className="pb-3 px-3"
-        classNames={{
-          inputWrapper: "bg-neutral-700 border border-neutral-600 rounded-lg",
-          input: "pr-16 text-neutral-400",
-        }}
-        maxRows={10}
-        minRows={1}
-        variant="bordered"
-      />
+    <>
+      {currentTaskState === AgentState.AWAITING_USER_INPUT && (
+        <div className="relative">
+          <div className="absolute bottom-2 left-0 right-0 flex items-center justify-center">
+            <button
+              type="button"
+              className="relative border-1 text-xs rounded px-2 py-1 border-neutral-600 bg-neutral-700 cursor-pointer select-none"
+              onClick={handleSendContinueMsg}
+            >
+              <div className="flex items-center">
+                <IoIosRefresh className="inline mr-2 w-2.5 h-2.5" />
+                <span className="inline-block">
+                  {t(I18nKey.CHAT_INTERFACE$INPUT_CONTINUE_MESSAGE)}
+                </span>
+              </div>
+            </button>
+          </div>
+        </div>
+      )}
```

Comment:
The `<ChatInput>` component contains logic related to user input for the chat. This code block seems out of place as it has nothing to do with the user input. Maybe you can move this logic to the parent, above `<ChatInput>`? That would also remove the dependency on `currentTaskState`. (imagine you are going to reuse this component in another part of the app, does it make sense to also reuse this "continue" logic you have implemented here?)

Reply:
I've considered both situations. At first, I thought the "continue button" was more like a quick reply linked to the `<ChatInput />`. After looking into it more carefully, I agree that the way you mentioned makes more sense :)